client-java/src/unit-tests/java/uk/gov/gchq/palisade/client/download/DownloadManagerTest.java

```diff
@@ -0,0 +1,258 @@
+        assertThat(scheduledEvent.getResource()).isEqualTo(resource);
+
+        eventObject = events.get(1);
+        assertThat(eventObject).isInstanceOf(DownloadStartedEvent.class);
+        var startedEvent = (DownloadStartedEvent) eventObject;
+        assertThat(startedEvent.getId()).isEqualTo(downloadId);
+        assertThat(startedEvent.getResource()).isEqualTo(resource);
+
+        // the second event caught should be a download completed
+
+        eventObject = events.get(2);
+        assertThat(eventObject).isInstanceOf(DownloadCompletedEvent.class);
+        var completedEvent = (DownloadCompletedEvent) eventObject;
+        assertThat(completedEvent.getId()).isEqualTo(downloadId);
+        assertThat(completedEvent.getResource()).isEqualTo(resource);
+
+    }
+
+    @Test
+    void testFailedMissingDownload() {
+
+        var resourceName = "i do not exist";
+
+        var resource = createResource(r -> r
+            .leafResourceId(resourceName)
+            .token(TOKEN)
+            .url(BASE_URL));
+
+        var downloadId = downloadManager.schedule(
+            resource,
+            eventBus,
+            new FileReceiver(),
+            configuration);
+
+        await().atMost(ofSeconds(5)).until(() -> events.size() == 3);
+        assertThat(events).hasSize(3);
+
+        // the first event caught should be a download scheduled
+
+        var eventObject = events.get(0);
+        assertThat(eventObject).isInstanceOf(DownloadScheduledEvent.class);
+        var scheduledEvent = (DownloadScheduledEvent) eventObject;
+        assertThat(scheduledEvent.getId()).isEqualTo(downloadId);
+        assertThat(scheduledEvent.getResource()).isEqualTo(resource);
+
+        // the second event caught should be a download started
+
+        eventObject = events.get(1);
+        assertThat(eventObject).isInstanceOf(DownloadStartedEvent.class);
+        var startedEvent = (DownloadStartedEvent) eventObject;
+        assertThat(startedEvent.getId()).isEqualTo(downloadId);
+        assertThat(startedEvent.getResource()).isEqualTo(resource);
+
+        // the second event caught should be a download completed
+
+        eventObject = events.get(2);
+        assertThat(eventObject).isInstanceOf(DownloadFailedEvent.class);
+        var failedEvent = (DownloadFailedEvent) eventObject;
+        assertThat(failedEvent.getId()).isEqualTo(downloadId);
+        assertThat(failedEvent.getResource()).isEqualTo(resource);
+        assertThat(failedEvent.getCause()).isInstanceOf(DownloaderException.class);
+        var exception = (DownloaderException) failedEvent.getCause();
+        assertThat(exception.getStatusCode()).isEqualTo(404);
+        assertThat(exception).hasMessage("Resource \"" + resourceName + "\" not found").hasNoCause();
+
+    }
+
+    /**
+     * Handles the provided event. Any method annotated with {@code Subscribe} must
+     * be public and be in a plublic class. This is a requirement of the
```

Comment:
spelling of plublic -> public, same on 212, 224, 236, 248

Reply:
fixed